plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergReadVersionedTable.java

```diff
@@ -93,6 +94,18 @@ public void testEndVersionInTableNameAndForClauseShouldFail()
                 "Cannot specify end version both in table name and FOR clause");
     }
 
+    @Test
+    public void testSelectFromSystemTablesWithEndSnapshotId()
+    {
+        assertQuery("SELECT * FROM \"test_iceberg_read_versioned_table$data\" FOR VERSION AS OF " + v1SnapshotId, "VALUES ('a', 1)");
+        assertQuerySucceeds("SELECT * FROM \"test_iceberg_read_versioned_table$properties\" FOR VERSION AS OF " + v1SnapshotId);
+        assertQueryFails("SELECT * FROM \"test_iceberg_read_versioned_table$history\" FOR VERSION AS OF " + v1SnapshotId, "Snapshot ID not supported for history table\\: tpch.test_iceberg_read_versioned_table\\$history");
+        assertQueryFails("SELECT * FROM \"test_iceberg_read_versioned_table$snapshots\" FOR VERSION AS OF " + v1SnapshotId, "Snapshot ID not supported for snapshots table\\: tpch.test_iceberg_read_versioned_table\\$snapshots");
+        assertEquals(computeActual("SELECT * FROM \"test_iceberg_read_versioned_table$files@"+ v1SnapshotId+"\"").getRowCount(), 1);
```

Comment:
The query verifying that the first  `INSERT` into a table produces only 1 data file works as expected:
```
        assertEquals(computeActual("SELECT * FROM \"test_iceberg_read_versioned_table$files@"+ v1SnapshotId+"\"").getRowCount(), 1);
```

The same query rewritten with `FOR VERSION AS OF`  does not work because it queries the latest version of the system table (corresponding to `v2SnapshotId`):

```
        assertEquals(computeActual("SELECT * FROM \"test_iceberg_read_versioned_table$files\" FOR VERSION AS OF " + v1SnapshotId).getRowCount(), 1);
```


The first query works because it transports the `snapshotId` of the system table into the table name which allows the correct instantiation of the `FilesTable` even when using the call `io.trino.metadata.Metadata#getSystemTable(io.trino.Session, io.trino.metadata.QualifiedObjectName)` for retrieving the system table.

The snapshot information from `FilesTable` is lost within the method: `io.trino.connector.system.SystemTablesMetadata#getTableHandle(io.trino.spi.connector.ConnectorSession, io.trino.spi.connector.SchemaTableName, java.util.Optional<io.trino.spi.connector.ConnectorTableVersion>, java.util.Optional<io.trino.spi.connector.ConnectorTableVersion>)`

where only the table name is used to obtain the `SystemTableHandle`:

```
SystemTableHandle.fromSchemaTableName(tableName)
```

**To be discussed**: Probably there is needed some extra information in the `SystemTableHandle` class to carry the table version.